src/category_theory/preadditive.lean

```diff
@@ -79,21 +79,21 @@ mk' (λ g, f ≫ g) $ λ g g', by simp
 def right_comp (P : C) {Q R : C} (g : Q ⟶ R) : (P ⟶ Q) →+ (P ⟶ R) :=
 mk' (λ f, f ≫ g) $ λ f f', by simp
 
-@[simp] lemma sub_comp {P Q R : C} (f f' : P ⟶ Q) (g : Q ⟶ R) :
+@[simp, reassoc] lemma sub_comp {P Q R : C} (f f' : P ⟶ Q) (g : Q ⟶ R) :
   (f - f') ≫ g = f ≫ g - f' ≫ g :=
 map_sub (right_comp P g) f f'
 
-@[simp] lemma comp_sub {P Q R : C} (f : P ⟶ Q) (g g' : Q ⟶ R) :
+@[reassoc, simp] lemma comp_sub {P Q R : C} (f : P ⟶ Q) (g g' : Q ⟶ R) :
   f ≫ (g - g') = f ≫ g - f ≫ g' :=
 map_sub (left_comp R f) g g'
```

Comment:
Does the attribute order matter here? I suspect it does, and as I can't see a reason why you wouldn't want to put `simp` on the `reassoc` lemmas, if this is intentional there probably should be a comment explaining it.

Reply:
Yup, afaict the order matters, and `reassoc` should come last.

Reply:
Yes, the order matters, but if the reassoc version is a simp lemma, then the redundant simp lemma linter complains that simp can prove the reassoc version. I have added a comment to this effect, but I could add a nolint declaration instead, I guess? Which do you prefer?

Reply:
Comment is better, thanks!